ReactNativeClient/lib/components/screens/config.js

```diff
@@ -30,11 +31,48 @@ class ConfigScreenComponent extends BaseScreenComponent {
 			await shared.checkSyncConfig(this, this.state.settings);
 		}
 
-		this.saveButton_press = () => {
+		this.saveButton_press = async () => {
+			if (
+				this.state.changedSettingKeys.includes('sync.target')
+				&& this.state.settings['sync.target'] === SyncTargetRegistry.nameToId('filesystem')
+				&& !await this.checkFilesystemPermission()
+			) {
+				Alert.alert(
+					_('Permission error'),
```

Comment:
Please name it something like "Sync target error" otherwise it looks like it's the settings that cannot be written to disk due to a permission error.